### DISCUSSION TITLE
New Ender 3 v2

### DISCUSSION THREAD

**Discussion by JimVandy85** [Jyers/Marlin#1309] [Q&A] [answered]
Recently got a ender 3 v2 from a friend and have done some upgrades before initial setup. Below are my upgrades. I'm running the 4.2.2 main board with the most recent version of jyers framework.

- HICTOP Dual z-axis kit
- light compression springs
- creality CR touch
- creality aluminum extruder upgrade
- capricorn tubing
- 0.2mm nozzle

So here's the issue, after getting everything assembled and installed, I go to home the z-axis. When I do this, the cr touch works just like it should as it approaches the bed. The probe stops the z-axis movement as it touches. Now. When it travels up the z-axis, the gantry its the top of the printer. What do I need to do to adjust the upper limit of the zaxis? I've attached a couple pics of my setup. 

Thanks and Best Wishes
![20210814_132209](https://user-images.githubusercontent.com/88944018/129456299-817b5883-ddb6-40f4-a8ba-4c310267bd67.jpg)
![20210814_132154](https://user-images.githubusercontent.com/88944018/129456314-d10ff86e-6481-4c69-83e4-a7a0a3515cf3.jpg)

**Comment by Cal-LF** [comment]
Change Z height in the slicer should keep it from limiting out. Other option is to change the bed size in the firmware which will require compiling your own firmware. 
Joining Jyers patreon automated compiling service will allow you to order custom firmware. https://www.patreon.com/jyers

**Reply by JimVandy85** [reply]
Whats a good compiler for the jyers fw?

**Comment by Cal-LF** [comment, marked as answer]
MS VS Code is what I have used. I don't know of any other. I followed Teaching Tech video and a couple other videos to get going with it.
Download VS Code https://code.visualstudio.com/download
Next install Platform IO extension and Auto Build Marlin extension to complete. 
https://www.youtube.com/watch?v=eq_ygvHF29I  2:29 time start to 7:30 time stop, for pertinent info.

Use one of Jyers E3V2 Templates copied to the Marlin folder as a starting point.
#define Z_MAX_POS 250 in configuation_h will be the one to change. Around line 1360.

**Reply by JimVandy85** [reply]
> Rather #define Z_MAX_POS 240, because of the upper supports of the Z axes ...
>  @JimVandy85 
>  Also provide synchronization of the 2 axes by a 618 mm GT2 belt (or 630 mm with a tensioner) and increase the vRef of the Z stepper to 1.22v (because 2 steps motor) ...

Is the belt necessary since the stepper motors are not ran individually? They are connected via a split harness off of the main z-axis. Forgive me, I don't know all of the correct terminology.

**Reply by vali20** [reply]
Interesting. Do you have a link where they can be bought? I'm interested in your version

**Reply by vali20** [reply]
Yes, you went a much better way than I did. I bought the version with stepper, which came with a screw identical to the original one, but which are shorter than the ones in your kit. So, now I have to change both the screws. Thanks for the reply!

**Reply by FinlessBob** [reply]
Why even use 2 steppers then?  I have the belt setup only and it's better than 2 steppers in my opinion as you do not need to worry about syncing them.

**Reply by vali20** [reply]
Because of the weight and not to overload the stepper

**Reply by vali20** [reply]
Can anyone tell me the length of the screws in your kit?

**Reply by pmaersk** [reply]
@tititopher68-dev do you have link to your Ender 3 V2 hotend housing. I find something like that on Thingiverse, but yours look better. Thanks in advance.
